cfgov/paying_for_college/models/pages.py

```diff
@@ -1,4 +1,56 @@
-# -*- coding: utf-8 -*-
-# from __future__ import unicode_literals
+from __future__ import unicode_literals
 
-# home for future wagtail page models
+from wagtail.wagtailadmin.edit_handlers import (
+    ObjectList, StreamFieldPanel, TabbedInterface
+)
+from wagtail.wagtailcore import blocks
+from wagtail.wagtailcore.fields import StreamField
+
+from paying_for_college.models.blocks import GuidedQuiz
+
+from v1.atomic_elements import molecules, organisms
+from v1.models import CFGOVPage, CFGOVPageManager
+
+
+class StudentResourcesPage(CFGOVPage):
+    """A page to serve subpages of paying-for-college."""
+    header = StreamField([
+        ('text_introduction', molecules.TextIntroduction()),
+        ('featured_content', organisms.FeaturedContent()),
+    ], blank=True)
+
+    content = StreamField([
+        ('full_width_text', organisms.FullWidthText()),
+        ('guided_quiz', GuidedQuiz()),
+        ('info_unit_group', organisms.InfoUnitGroup()),
+        ('expandable_group', organisms.ExpandableGroup()),
+        ('expandable', organisms.Expandable()),
+        ('well', organisms.Well()),
+        ('raw_html_block', blocks.RawHTMLBlock(
+            label='Raw HTML block'
+        )),
+    ], blank=True)
+
+    content_panels = CFGOVPage.content_panels + [
+        StreamFieldPanel('header'),
+        StreamFieldPanel('content'),
+    ]
+    # Tab handler interface
+    edit_handler = TabbedInterface([
+        ObjectList(content_panels, heading='General Content'),
+        ObjectList(CFGOVPage.sidefoot_panels, heading='Sidebar'),
+        ObjectList(CFGOVPage.settings_panels, heading='Configuration'),
+    ])
+    objects = CFGOVPageManager()
+
+    def get_template(self, request):
+        for block in self.content:
+            block.value['situation_id'] = block.id
+        return 'paying-for-college/{}.html'.format(
+            self.slug)
```

Comment:
+1 for having a design discussion about this, if you like @higs4281.

Reply:
I would love to discuss. We were trying to avoid type proliferation, but I can see the downsides.

Reply:
I created three page types with explicit template assignments to handle our page needs.

Reply:
Thanks, @higs4281. This implementation duplicates a lot between the three page types: basically everything instead of the content StreamField blocks and the template. What would you think about introducing an [`abstract`](https://github.com/wagtail/wagtail/blob/e844200f27fe6a2089175b488a283bc27331c4c6/wagtail/contrib/routable_page/models.py#L138) parent page type for these, with three concrete models that inherit from it and only define the template code, and content field? This would reduce a lot of duplication. For the content field you could even do something like we have [here](https://github.com/cfpb/cfgov-refresh/blob/master/cfgov/v1/models/browse_filterable_page.py#L19), for example:

```py
class PayingForCollegePage(CFGOVPage):
    header = ...

    ...

    class Meta:
        abstract = True

class PayingForCollegeContent(StreamBlock):
    full_width_text = organisms.FullWidthText()
    info_unit_group = organisms.InfoUnitGroup()
    expandable_group = organisms.ExpandableGroup()
    expandable = organisms.Expandable()
    well = organisms.Well()
    raw_html_block = blocks.RawHTMLBlock(label='Raw HTML block')

class RepayingStudentDebtPage(PayingForCollegePage):
    content = StreamField(PayingForCollegeContent, blank=True)
    template_name = 'paying-for-college/repaying-student-debt.html'

class CollegeCostsPage(PayingForCollegePage):
    content = StreamField(PayingForCollegeContent, blank=True)
    template_name = 'paying-for-college/college-costs.html'

class StudentLoanQuizContent(PayingForCollegeContent):
    guided_quiz = GuidedQuiz()

class StudentLoanQuizPage(PayingForCollegePage):
    content = StreamField(StudentLoanQuizContent, blank=True)
    
    def get_template(self):
        ...
```

Unfortunately Django model inheritance doesn't let you redefine a field on a parent model, the way vanilla Python class inheritance does; this prevents you from defining the base `content` on the parent and overriding as needed in the child.